src/server/pfs/server/driver_v2.go

```diff
@@ -536,3 +581,81 @@ func (d *driverV2) globFileV2(pachClient *client.APIClient, commit *pfs.Commit,
 		return f(fr.Info())
 	})
 }
+
+func (d *driverV2) copyFile(pachClient *client.APIClient, src *pfs.File, dst *pfs.File, overwrite bool) (retErr error) {
+	ctx := pachClient.Ctx()
+	if overwrite {
+		// TODO: after delete merging is sorted out add overwrite support
+		return errors.New("overwrite not yet supported")
+	}
+	srcPrefix := cleanPath(src.Path)
+	dstPrefix := cleanPath(dst.Path)
+	pathTransform := func(x string) string {
+		if !strings.HasPrefix(x, srcPrefix) {
+			panic("cannot apply path transform")
+		}
+		return path.Join(dstPrefix, x[len(srcPrefix):])
+	}
+
+	mr, err := d.storage.NewMergeReader(ctx, []string{commitKey(src.Commit)})
```

Comment:
This is using a helper function from v1 for creating the commit prefix. It is correct because the etcd prefix in v1 is the same as the object storage prefix in v2, but I think it would be better if we were to use this in the other places where we compute a commit prefix for consistency.